### PR TITLE
[feature] add matrixbuilds with --prefer-lowest and default dependency versions

### PR DESCRIPTION
**Pull request by verfriemelt-dot-org**
to ensure compatibility with lower versions we should run tests with `--prefer-lowest` and switch to `composer update` to install dependencies.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -58,13 +58,14 @@ jobs:
         run: make analyse
 
   tests:
-    name: "Tests on PHP ${{ matrix.php-version }}"
+    name: "Tests on PHP ${{ matrix.php-version }} ${{ matrix.prefer-lowest }}"
```
Comment (verfriemelt-dot-org) [inline .github/workflows/build.yaml:57]: this could be more concise 
like this for example:
```
testsuite (8.1, lowest)
```

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,4 @@ phpunit.xml
 composer.lock
 /coverage
 /vendor
+/.idea
```
Comment (verfriemelt-dot-org) [inline .gitignore]: ups

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -1,10 +1,6 @@
 on:
   pull_request:
   push:
-    branches:
-      - main
-    tags:
-      - "**"
```
Comment (jschaedl) [inline .github/workflows/build.yaml:7]: 👍🏻

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -31,7 +27,7 @@ jobs:
           coverage: none
 
       - name: "Install dependencies"
-        run: composer install --no-interaction --no-progress --no-suggest --optimize-autoloader
+        run: composer install --no-interaction --no-progress --optimize-autoloader
```
Comment (jschaedl) [inline .github/workflows/build.yaml:30]: ah yes, thanks

File: .github/workflows/build.yaml (hunk 1 above)
Comment (jschaedl) [reply .github/workflows/build.yaml:57]: feel free to change it

**Review by jschaedl — APPROVED** [review]

**Comment by jschaedl** [comment]
Not sure why some tests are failing, but it's suspicious that those are the one's needing env vars 🤔

**Comment by jschaedl** [comment]
Going to merge here and figure out a solution for the failing tests afterwards.

Thank you @verfriemelt-dot-org!

**Pull request merged**